### PR TITLE
feat: upload models from local to S3 directly

### PR DESCRIPTION
**Pull request by o0OooO**
## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

截屏2023-08-07 13 49 43

### CODE REVIEW

**Comment by alvindaiyan** [comment]
What's the maximum file size and number for this feature?

**Review by alvindaiyan — CHANGES_REQUESTED** [review]

File: scripts/main.py [hunk 1]
```diff
@@ -783,95 +783,94 @@ def update_api_key():
...
-                                                             choices=sorted(scan_textural_inversion_model()),
-                                                             elem_id="textual_inversion_model_dropdown")
-                        create_refresh_button(textual_inversion_path, scan_textural_inversion_model,
-                                              lambda: {"choices": sorted(scan_textural_inversion_model())},
-                                              "refresh_textual_inversion_model")
+                        model_type_drop_down = gr.Dropdown(label="Model Type", choices=["SD Checkpoints", "Textual Inversion", "LoRA model", "ControlNet model", "Hypernetwork", "VAE"], elem_id="model_type_ele_id")
+                        model_type_hiden_text = gr.Textbox(elem_id="model_type_value_ele_id", visible=False)
+                        def change_model_type_value(model_type: str):
+                            model_type_hiden_text.value = model_type
+                            print(f"change_model_type_value:{model_type}")
```
Comment (yike5460) [inline scripts/main.py:858]: use console.log instead

File: aws_extension/sagemaker_ui.py [hunk 2]
```diff
@@ -339,115 +339,115 @@ def refresh_all_models():
...
-        except Exception as e:
-            print(f"fail to upload model {lp}, error: {e}")
-
-    print(f"Refresh checkpionts after upload...")
-    refresh_all_models()
-
-    return log, None, None, None, None, None
+    log = "start upload model to s3:"
+
+    # local_paths = [sd_checkpoints_path, textual_inversion_path, lora_path, hypernetwork_path, controlnet_model_path]
```
Comment (yike5460) [inline aws_extension/sagemaker_ui.py:344]: will such code block add back, delete if not

File: aws_extension/sagemaker_ui.py [hunk 3]
```diff
@@ -339,115 +339,115 @@ def refresh_all_models():
...
+    #         print(response)
+    #
+    #         log = f"\n finish upload {local_tar_path} to {s3_base}"
+    #
+    #         # os.system(f"rm {local_tar_path}")
+    #         rm(local_tar_path, recursive=True)
+    #     except Exception as e:
+    #         print(f"fail to upload model {lp}, error: {e}")
+    #
+    # print(f"Refresh checkpionts after upload...")
```
Comment (yike5460) [inline aws_extension/sagemaker_ui.py:447]: typo, checkpoints

File: javascript/txt2img-config.js [hunk 4]
```diff
@@ -34,7 +34,208 @@ window.onload = function() {
     }, 2000);
 };
 
+let uploadedFilesMap = new Map();
+let chunkSize = 1024 * 1024 * 1024; // 1GB chunk size, you can adjust this as needed.
```
Comment (yike5460) [inline javascript/txt2img-config.js:38]: no magic number

**Review by yike5460 — COMMENTED** [review]
make sure used/redundant code removed

**Pull request merged**